demisto_sdk/commands/update_release_notes/update_rn_manager.py

```diff
@@ -62,29 +62,65 @@ def manage_rn_update(self):
             print_color('\nSuccessfully updated the following packs:\n' + '\n'.join(self.total_updated_packs),
                         LOG_COLORS.GREEN)
 
-    def get_git_changed_files(self) -> Tuple[set, set, set, set]:
+    def filter_to_relevant_files(self, file_set: set, validate_manager: ValidateManager) -> Tuple[set, set]:
+        """
+        Given a file set, filter it to only files which require RN and if given, from a specific pack
+        """
+        filtered_set = set()
+        if self.given_pack:
+            for file in file_set:
+                if isinstance(file, tuple):
+                    file_path = str(file[1])
+
+                else:
+                    file_path = str(file)
+
+                file_pack_name = get_pack_name(file_path)
+                if file_pack_name not in self.given_pack:
+                    continue
+
+                filtered_set.add(file)
+
+        return validate_manager.filter_to_relevant_files(filtered_set)
```

Comment:
I think we can change this function to use validate_manager, it seems like it does some duplicate code. Or maybe even add a general function for filtering files by given pack so it will be generalized and can be called from other places in the future.

Also, this function having same name as validate_manager function is a little confusing, maybe filter_relevant_files_in_pack or something like that.
```suggestion
        filtered_set = validate_manager.filter_to_relevant_files(filtered_set)
        if self.given_pack:
            filtered_set = {file_path for file_path in filtered_set if get_pack_name(file_path) in self.given_pack}
        return filtered_set
```

Reply:
Here is the issue with this idea (which was my original direction) - this does not solve the "bug" element of this issue: https://github.com/demisto/etc/issues/39422

if we use validate_manager's filter command out the back it will try and look at the file_type of all files and only then filter the relevant files by path - hence it could fail on unrelated files.
to avoid this, I've moved the path filtration before the VM's filtration - might look not as clean but it is correct.

Reply:
Got it. Resolving